include/pybind11/detail/common.h

```diff
@@ -154,6 +154,11 @@
 // C4505: 'PySlice_GetIndicesEx': unreferenced local function has been removed (PyPy only)
 #  pragma warning(disable: 4505)
 #  if defined(_DEBUG) && !defined(Py_DEBUG)
+// fix for vs2022: https://github.com/pybind/pybind11/issues/3477
+#    include <yvals.h>
+#    if _MSVC_STL_VERSION >= 143
+#      include <crtdefs.h>
```

Comment:
A comment to explain that this is knowingly violating the Python.h include order would be good. For example:

```
// Workaround for a VS 2022 issue.
// NOTE: This workaround knowingly violates the Python.h include order requirement:
// https://docs.python.org/3/c-api/intro.html#include-files
// See https://github.com/pybind/pybind11/pull/3497 for full context.
```

Reply:
Before I wrote:  From what I see it's not gated on PY_VERSION_HEX or the value of _MSC_VER.

I see it now, it's already gated on Python <= 3.5. Sorry I missed it because the github view didn't show it.